### DISCUSSION TITLE
Component Request: Drawer Component

### DISCUSSION THREAD

**Discussion by pevey** [melt-ui/melt-ui#193] [Ideas]
I'd also like to request consideration of a Drawer component. Overlay that comes out from window edge, configurable left/right/top/bottom.

I'm super excited to see this project coming along.  I can't wait to convert some projects.

EDIT:  Adding this link for reference as it may be insightful: https://github.com/radix-ui/primitives/issues/986
Summary: Drawer can be built in Radix on top of the Dialog primitive.  Radix devs decided not to make it separate component for now.  The decision does not seem to have been super popular, but I can see their reasoning.

**Comment by TGlide** [comment]
Hi! In Melt, we have an example of the Drawer being built from the Dialog as well. Since positioning is not controlled from within Melt, I also think it doesn't make sense to have a separate drawer builder.

https://www.melt-ui.com/docs/builders/dialog#drawer

**Reply by pevey** [reply]
Oh, that will work well. Thanks.
